gobblin-service/src/test/java/org/apache/gobblin/service/modules/orchestration/DagManagerTest.java

```diff
@@ -943,30 +162,19 @@ public void testDagManagerQuotaExceeded() throws URISyntaxException, IOException
         .thenReturn(jobStatusIteratorFlow1_0)
         .thenReturn(jobStatusIteratorFlow1_1);
 
-    this._dagManagerThread.run();
     // dag will not be processed due to exceeding the quota, will log a message and exit out without adding it to dags
-    this.queue.offer(dagList.get(1));
-    this._dagManagerThread.run();
     SortedMap<String, Counter> allCounters = metricContext.getParent().get().getCounters();
     Assert.assertEquals(allCounters.get(MetricRegistry.name(
         ServiceMetricNames.GOBBLIN_SERVICE_PREFIX,
         ServiceMetricNames.SERVICE_USERS,
         "user")).getCount(), 1);
-
-    this._dagManagerThread.run(); // cleanup
-
-    Assert.assertEquals(this.dags.size(), 0);
-    Assert.assertEquals(this.jobToDag.size(), 0);
-    Assert.assertEquals(this.dagToJobs.size(), 0);
   }
 
-  @Test (dependsOnMethods = "testDagManagerQuotaExceeded")
-  public void testQuotaDecrement() throws URISyntaxException, IOException {
-
-    List<Dag<JobExecutionPlan>> dagList = buildDagList(3, "user", ConfigFactory.empty());
+  @Test (dependsOnMethods = "testDagManagerQuotaExceeded", enabled = false)
+  // todo re-write for dag proc
+  public void testQuotaDecrement() throws URISyntaxException {
+    List<Dag<JobExecutionPlan>> dagList = DagTestUtils.buildDagList(3, "user", ConfigFactory.empty());
```

Comment:
1. do these tests for the quota manager still belong in the `DagManagerTest` class?
2. is there a more direct way to write these tests?  I'm not seeing which method of the quota mgr is invoked.  also the test assertions are based upon counters.

Reply:
imo, they should be moved to QuotaManagerTest class.
I think in https://github.com/apache/gobblin/pull/3481/files#diff-de0fafb1b3f630945d2bee38692233874e775d089d3737a4ad19d9d0abd729bf they were written in this class, because it involves running DagManagerThread and DMT invokes quota manager APIs, and this is also the reason no quota manager apis can be seen invoked right in the tests.
I think, we can distribute these tests in DagProc tests and QuotaManagerTests. In QuotaManager test, we can invoke the APIs and assert on the responses. And in DagProc tests, we can assert the number of calls to QuotaManager apis.